spring-batch-core/src/main/java/org/springframework/batch/core/metrics/BatchMetrics.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.batch.core.metrics;
+
+import java.util.Arrays;
+import java.util.function.Supplier;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.LongTaskTimer;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
+
+/**
+ * Main entry point to interact with Micrometer's {@link Metrics#globalRegistry}.
+ * Provides common metrics such as {@link Timer}, {@link Counter} and {@link Gauge}.
+ *
+ * Only intended for internal use.
+ *
+ * @author Mahmoud Ben Hassine
+ */
+public final class BatchMetrics {
+
+	private static final String METRICS_PREFIX = "spring.batch.";
+
+	public static final String STATUS_SUCCESS = "SUCCESS";
+
+	public static final String STATUS_FAILURE = "FAILURE";
+
+	private BatchMetrics() {}
+
+	/**
+	 * Create a {@link Timer}.
+	 *
+	 * @param name        of the timer. Will be prefixed with {@link BatchMetrics#METRICS_PREFIX}.
+	 * @param description of the timer
+	 * @param tags        of the timer
+	 * @return a new timer instance
```

Comment:
I'm not saying we need to make a big deal out of it right now, but your column based layout for the `@param` javadoc is different than the rest of the codebase.  We should come to an agreement on one way or the other.

Reply:
Fixed to be consistent with the rest of the code base.